### DISCUSSION TITLE
Suggested Improvement for those running shared dotfiles systems

### DISCUSSION THREAD

**Discussion by ethereal-engineer** [pyenv/pyenv#2784] [Show and tell]
```bash
# pyenv if pyenv is installed
export PYENV_ROOT="$HOME/.pyenv"
command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"
command -v pyenv >/dev/null && eval "$(pyenv init -)"
```
Here the addition of the conditional check for the pyenv command a second time before calling eval will prevent report of pyenv not found on systems that do not have it installed, but allow the init to work on systems that do.

**Comment by native-api** [comment]
Shouldn't the other 2 commands also be conditional on Pyenv being installed then?

**Comment by native-api** [comment]
E.g. the 2nd command blindly adds to PATH if `pyenv` is not on PATH since `$PYENV_ROOT/bin` is the standard off-the-PATH location.
So despite the condition, it also assumes that Pyenv is installed.

**Comment by ethereal-engineer** [comment]
Ideally yes, but a single non-existing directory in the PATH environment variable is quickly dropped by the assessor of the PATH, so it is low-impact. Keeping in mind that PATH is used only for initial execution of binary/script. I made a version that did a conditional check for all, but this version is the least amount of change to the original, and the most syntactically compatible across all supported shells.

**Comment by ethereal-engineer** [comment]
I mean, this could easily be improved upon by adding a second test on the first "command" line that checks for the existence of the pyenv script before modifying the PATH, using `test -e`. But this wasn't so much a be all and end all as much as it was a "here - I needed this improvement - someone might find it useful - or maybe someone could make it better".
